lib/internal/modules/esm/resolve.js

```diff
@@ -1154,14 +1162,14 @@ function defaultResolve(specifier, context = {}) {
  */
 function decorateErrorWithCommonJSHints(error, specifier, parentURL) {
   const found = resolveAsCommonJS(specifier, parentURL);
-  if (found) {
+  if (found && found !== specifier) {
     // Modify the stack and message string to include the hint
-    const lines = StringPrototypeSplit(error.stack, '\n');
-    const hint = `Did you mean to import ${found}?`;
+    const endOfFirstLine = StringPrototypeIndexOf(error.stack, '\n');
```

Comment:
Nit: Splitting and later splicing might be a little bit easier to understand.

Reply:
But it's also vulnerable to prototype pollution (and we're inside the error path)

https://github.com/nodejs/node/blob/b0443690f89816ca3b9b8dee7e73f4527b3ff3e0/doc/contributing/primordials.md#L619-L628